dynamic_sampling_context.go

```diff
@@ -36,10 +37,57 @@ func NewDynamicSamplingContext(header []byte) (DynamicSamplingContext, error) {
 	}, nil
 }
 
+func DynamicSamplingContextFromTransaction(span *Span) (DynamicSamplingContext, error) {
+	entries := map[string]string{}
+
+	hub := hubFromContext(span.Context())
+	scope := hub.Scope()
+	options := hub.Client().Options()
+
+	if traceID := span.TraceID.String(); traceID != "" {
+		entries["trace_id"] = traceID
+	}
+	if sampleRate := span.sampleRate; sampleRate != 0 {
+		entries["sample_rate"] = strconv.FormatFloat(sampleRate, 'f', -1, 64)
+	}
+
+	if dsn := hub.Client().dsn; dsn != nil {
```

Comment:
l: can we declare `client := hub.Client()` above and re-use it for both the options call and the dsn call?

Reply:
https://github.com/getsentry/sentry-go/pull/492/commits/b182fc54e4fa80621648d2a6b006b359593b135b